File: src/main/resources/archetype-resources/fabric/src/main/java/integration/ModMenuIntegration.java
package ${package}.integration;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

public final class ModMenuIntegration implements ModMenuApi {

    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        if (!${modName}Fabric.isClothConfigLoaded())
            return super.getModConfigScreenFactory();
        return (parent) -> ${modName}Fabric.getConfigScreen(parent);
    }
}

File: src/main/resources/archetype-resources/fabric/src/main/kotlin/integration/ModMenuIntegration.kt
package ${package}.integration

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi

import ${package}.${modName}Fabric

class ModMenuIntegration : ModMenuApi {

    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        if (!${modName}Fabric.isClothConfigLoaded())
            return super.getModConfigScreenFactory()
        return ConfigScreenFactory { ${modName}Fabric.getConfigScreen(it) }
    }
}
